### PR TITLE
Remove jQuery from component lifecycle

### LINKED ISSUE #197: Remove the use of jQuery in code samples for didInsertElement in components/the-component-lifecycle.md
https://github.com/ember-learn/guides-source/blob/master/guides/v3.4.0/components/the-component-lifecycle.md#integrating-with-third-party-libraries-with-didinsertelement

### LINKED ISSUE #198: Remove the use of jQuery in code samples for didRender in components/the-component-lifecycle.md
https://github.com/ember-learn/guides-source/blob/master/guides/v3.4.0/components/the-component-lifecycle.md#making-updates-to-the-rendered-dom-with-didrender

### LINKED ISSUE #199: Remove the use of jQuery in code samples for willDestroyElement in components/the-component-lifecycle.md
https://github.com/ember-learn/guides-source/blob/master/guides/v3.4.0/components/the-component-lifecycle.md#detaching-and-tearing-down-component-elements-with-willdestroyelement

### PR DESCRIPTION
**Pull request by jkarsrud**
This fixes #197, fixes #198 and fixes #199.

I'm not 100% sure if we should be using the date picker example anymore, but I kept it as a straight conversion right now in order for the diffs to be easier to see and review.

### CODE REVIEW

File: guides/v3.4.0/components/the-component-lifecycle.md [hunk 1]
```diff
@@ -246,7 +247,7 @@ export default Component.extend({
 
   didRender() {
     this._super(...arguments);
-    this.$('.item-list').scrollTop(this.$('.selected-item').position().top);
+    this.element.querySelector('.item-list').scrollTop = this.element.querySelector('.selected-item').offsetTop);
```
Comment (kamry-bowman) [inline guides/v3.4.0/components/the-component-lifecycle.md]: I'm not sure this works, as offsetTop will determine use the document or nearest *positioned* document. Not sure you can assume that the .item-list would be positioned. Here is how I tackled this:
```
const scrollTarget = Math.abs(this.element.getBoundingClientRect().top - this.element.querySelector('.selected-item').getBoundingClientRect().top);
this.element.querySelector('.item-list').scrollTop = scrollTarget;   
```

File: guides/v3.4.0/components/the-component-lifecycle.md [hunk 2]
```diff
@@ -272,8 +273,8 @@ import Component from '@ember/component';
 
 export default Component.extend({
   willDestroyElement() {
-    this.$().off('animationend');
-    this.$('input.date').myDatepickerLib().destroy();
+    this.element.removeEventListener('animationend');
```
Comment (kamry-bowman) [inline guides/v3.4.0/components/the-component-lifecycle.md]: I think removeEventListener takes a second argument, which needs to be a function reference. This poses a problem, as we would need to have a reference to the original function we put on the element with addEventListener. Don't know how we want to handle that?

File: guides/v3.4.0/components/the-component-lifecycle.md [hunk 3]
```diff
@@ -200,7 +201,7 @@ There are a few things to note about the `didInsertElement()` hook:
   particularly when order of execution is important.
 
 [did-insert-element]: https://www.emberjs.com/api/ember/release/classes/Component/events/didInsertElement?anchor=didInsertElement
-[dollar]: https://www.emberjs.com/api/ember/release/classes/Component/methods/$?anchor=%24
+[element]: https://emberjs.com/api/ember/3.4/classes/Component/properties/element?anchor=element
```
Comment (jenweber) [inline guides/v3.4.0/components/the-component-lifecycle.md]: Can you change this link to use `/release/` instead of `/3.4/` in the url? Thanks!

**Comment by jenweber** [comment]
Hi! This PR was still open as of creating the 3.5 release. This means before this PR to be merged, any changes requested should be made, and then the same additions should be applied to the upcoming 3.5 version files. They will be available later this week. Sorry about the extra step!

**Comment by jkarsrud** [comment]
Don't worry about the extra step, I cause some extra steps for you when not commenting on the issues too! I'll make a Twiddle to show that it all works, and then wait for the 3.5 files to land.

File: guides/v3.4.0/components/the-component-lifecycle.md (hunk 2 above)
Comment (jkarsrud) [reply guides/v3.4.0/components/the-component-lifecycle.md]: You're absolutely right. I think we could get away with just creating the handler in the example that adds the event listener, and just using that in this sample. The samples in this file aren't *really* connected as a whole, so we can probably get away with some "cheating".

**Comment by jenweber** [comment]
@jkarsrud The 3.5 files are up. Thanks!

**Comment by jenweber** [comment]
@jkarsrud hey, do you still want to make these last changes and add them to `3.5` too?

**Comment by jkarsrud** [comment]
@jenweber Sorry about the delay. I've updated the exaples and copied the changes over to the 3.5.0 files. I have also made an Ember Twiddle that shows that these things work as expected: https://ember-twiddle.com/c8c9f62aa2bc9f7807edf89f8b246bfe?numColumns=2&openFiles=components.my-component.js%2Ctemplates.components.my-component.hbs

Let me know if there's anything else you want me to do in this PR!

**Comment by jkarsrud** [comment]
The Twiddle is updated with working examples for "everything" now. While implementing those examples, I noticed that the code actually didn't work as expected, but I'm not sure the fix is what we should be showing in these examples...

**Comment by jenweber** [comment]
This is awesome, thank you! I appreciate the Twiddles. Merged.

**Pull request merged**